src/v/net/probes.cc

```diff
@@ -21,76 +22,91 @@ namespace net {
 void server_probe::setup_metrics(
   ss::metrics::metric_groups& mgs, const char* proto) {
     namespace sm = ss::metrics;
+    auto aggregate_labels = config::shard_local_cfg().aggregate_metrics()
+                              ? std::vector<sm::label>{sm::shard_label}
+                              : std::vector<sm::label>{};
```

Comment:
nit: Would it be worth to create a simple helper function to return this `aggregate_labels` ?

Reply:
Sure, that would make sense. I think a good place to add this helper would be in`ssx/metrics.h` which i added in the [PR](https://github.com/redpanda-data/redpanda/pull/5165) for the new metrics endpoint. Maybe it would be best to wait for both to be merged in and add the helper in a follow up PR to minimise merge conflicts?

Reply:
If you cherry pick e7a353bb758e5d3e2d6a03da8080e68a5a4b9953 here, it should all merge fine.

Reply:
Assuming that the new metrics endpoint PR is merged first, would it skip the cherry picked commit when rebasing on dev?

Reply:
Yes. And also it will merge fine without a rebase.

Reply:
I thought we usually rebase on top of dev, instead of merging. I'll do it then.

Reply:
I've added an `aggregation_labels` function to `ssx/metrics.h` and called it in most places (minus one).
I didn't come up with something generic enough to work everywhere. I'm resolving the conversation, but
feel free to reopen if needed.